client/app/components/tagging/tagging.directive.js

```diff
@@ -163,35 +115,38 @@
         if (vm.tags.selectedTag.category && angular.isDefined(vm.tags.selectedTag.category.single_value)) {
           if (vm.tags.selectedTag.category.single_value) {
             // Find existing tag w/ category in tags.of_item
-            for (var i = 0; i < vm.tags.ofItem.length; i++) {
-              var tag = vm.tags.ofItem[i];
+            for (var i = 0; i < vm.tagsOfItem.length; i++) {
+              var tag = vm.tagsOfItem[i];
               if (tag.category.id === vm.tags.selectedTag.category.id) {
                 $scope.removeTag(tag);
                 break;
               }
             }
           }
         }
+
+        var matchingTag = $filter('filter')(vm.tagsOfItem, {id: vm.tags.selectedTag.id});
+
         // Add Selected Tag
-        if (vm.tags.ofItem.indexOf(vm.tags.selectedTag) === -1) {
-          saveOriginalTags();
-          vm.tags.ofItem.push(vm.tags.selectedTag);
+        if (!matchingTag.length) {
+          vm.tagsOfItem.push(getSmTagObj(vm.tags.selectedTag));
         }
       };
 
       $scope.removeTag = function(tag) {
```

Comment:
RIGHT!!!  I **LOVE** me some lodash, hope we can jump to 4.x at some point in the new future